griptape/tools/vector_store_client/tool.py

```diff
@@ -36,10 +38,12 @@ class VectorStoreClient(BaseTool):
             ),
         }
     )
-    def search(self, params: dict) -> BaseArtifact:
+    def search(self, params: dict) -> ListArtifact | ErrorArtifact:
         query = params["values"]["query"]
 
         try:
-            return self.query_engine.query(query, top_n=self.top_n, namespace=self.namespace)
+            entries = self.vector_store_driver.query(query, namespace=self.namespace, top_n=self.top_n)
```

Comment:
[base_vector_store_driver](https://github.com/griptape-ai/griptape/blob/41984d63beafe9c10c38a77bde2ea2a5cafa51f0/griptape/drivers/vector/base_vector_store_driver.py#L126) takes `count`, not `top_n`